src/rules.jl

```diff
@@ -184,13 +184,25 @@ DNERule(args...) = DNE()
 #####
 
 """
-TODO
+    WirtingerRule([𝒟::Type, ]P::AbstractRule, C::AbstractRule)
+Construct a `WirtingerRule` object, which is an `AbstractRule` that consists of
+an `AbstractRule` for both the primal derivative ``∂/∂x`` and the conjugate
+derivative ``∂/∂x̅``. If the domain `𝒟` is specified, return a `Rule` evaluating
+to `P(Δ) + C(Δ)` if `𝒟 <: Real`, otherwise return `WirtingerRule(P, C)`.
 """
 struct WirtingerRule{P<:AbstractRule,C<:AbstractRule} <: AbstractRule
     primal::P
     conjugate::C
 end
 
+function WirtingerRule(𝒟::Type, primal::AbstractRule, conjugate::AbstractRule)
```

Comment:
Lets make this an `AbstractRule`.
There is precident for declaring constructors for abstract types that return a suitable concrete subtype depending on their input,
and this makes me a lot less uncomfortable than having a constructor for `WirtingerRule` that does not always return a `WirtingerRule`

```suggestion
function AbstractRule(𝒟::Type, primal::AbstractRule, conjugate::AbstractRule)
```

Reply:
Yes! That's way better!

Reply:
Glad we could work that out, evantually.

Reply:
👍 
needs docstring(s) updating accordingly :)